File: Iode/Analysis/Lexical/TokenType.cs
namespace Iode.Analysis.Lexical
{
    /// <summary>
    /// All possible tokens
    /// </summary>
    public enum TokenType
    {
        // Types
        IDENTIFIER,
        NUMBER,
        BOOLEAN,
        STRING,
        NIL,

        // Mathematical
        ADD,
        SUB,
        MUL,
        DIV,

        // Syntactical
        COMMA,
        DOT,
        LPAREN,
        RPAREN,
        LBRACE,
        RBRACE,
        LBRACK,
        RBRACK,
        EQUALS,
        ARROW,

        // Keywords
        IF,
        ELSE,
        DEF,
        VAR,
        INT,
        STR,
        BOOL,

        // Special
        NEWLINE,
        VOID
    }
}

File: src/iode/scanning/TokenType.java
package iode.scanning;

public enum TokenType {
	// Regular	
	IDENTIFIER,
	STRING,
	NUMBER,
	BOOLEAN,
	
	// Keywords
	LET,
	VAR,
	FUNCTION,
	IMPORT,
	RETURN,
	
	// Symbols
	EQUALS,
	ADD,
	SUB,
	MUL,
	DIV,
	GT,
	LT,
	DOT,
	COMMA,
	LPAREN,
	RPAREN,
	LBRACE,
	RBRACE,
	LBRACK,
	RBRACK,
	NEWLINE
}
